Tests/AfluentTests/WaitUntilCondition.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  WaitUntilCondition.swift
+//  Afluent
+//
+//  Created by Annalise Mariottini on 11/9/24.
+//
+
+import Afluent
+
+/// Waits for some condition to proceeds, unless the specified timeout is reached, in which case an error is thrown.
+func wait(until condition: @autoclosure @escaping @Sendable () async -> Bool, timeout: Duration)
+    async throws
+{
+    try await DeferredTask {
```

Comment:
I have 2 suggestions. Let's make this accept a clock over using Task.sleep and let's have the duration we wait be until the next clock instant rather than a constant 100 nanoseconds.

Also, there may be a minor performance gain by ditching the task creation altogether and checking the timeout before/after the clock sleep. Now normally I wouldn't bring it up, but because the timeframe is so short and this could be a staple in low level functions, it might be worth looking at the minor improvements possible

Reply:
Made some improvements.

Also, just in case it isn't clear, this is just for tests.

Reply:
Thanks! I didn't catch that. Should we consider making it public? Maybe even as part of an AfluentTesting lib?  

I know I've reached for something like this a lot in my past

Reply:
That could be as a follow-up PR if at all

Reply:
Yeah, let's do in a follow up